### PR TITLE
test: init sharp integration test

### PR DESCRIPTION
**Pull request by ImBIOS**
### What does this PR do?

- #5852 

All 5 basic tests passing

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### How did you verify your code works?

I ran `bun test` locally

### CODE REVIEW

**Comment by paperclover** [comment]
thank you. sharp used to be a harder package to run because it depended on postinstall (see #7132 for status on that) but these should pass now.

File: test/integration/sharp/.gitignore [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# Based on https://raw.githubusercontent.com/github/gitignore/main/Node.gitignore
```
Comment (paperclover) [inline test/integration/sharp/.gitignore]: root of the entire project has a gitignore. please remove.

File: test/integration/sharp/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# bun-integration-test-sharp
```
Comment (paperclover) [inline test/integration/sharp/README.md]: this file can also be 💥

File: test/integration/sharp/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+console.log("Hello via Bun!");
```
Comment (paperclover) [inline test/integration/sharp/index.ts]: 💥

File: test/integration/sharp/package.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "bun-integration-test-sharp",
+  "module": "index.ts",
+  "type": "module",
+  "scripts": {
+    "postinstall": "cd node_modules/sharp && bun install/check.js"
```
Comment (paperclover) [inline test/integration/sharp/package.json]: this `postinstall` should be removed. if sharp doesnt work out of box then it is a bug in bun.

File: test/integration/sharp/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (paperclover) [inline test/integration/sharp/tsconfig.json]: 💥

**Review by paperclover — CHANGES_REQUESTED** [review]
please cleanup the generated files from bun init. they are not needed. all that is needed is package.json, sharp.test.ts, bun.lockb, and bun.png

**Comment by ImBIOS** [comment]
@paperdave updated, and I changed `postinstall` of sharp became `trustedDependencies`, is that correct?

**Comment by paperclover** [comment]
Ideally this test should function without any configuration, no `trustedDependencies` set, but that might depend on the linked PR. I guess in the meantime it can be merged if it passes, and we will remove the trustedDependencies flag

**Comment by paperclover** [comment]
doesnt pass now but we should be able to revisit these tests after postinstall branch merges and it hopefully will work as-is.

File: test/integration/sharp/sharp.test.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import { describe, expect, it } from "bun:test";
+import sharp from "sharp";
+
+describe("sharp integration tests", () => {
+  it("should resize an image", async () => {
+    const inputBuffer = await sharp("bun.png").resize(200, 200).toBuffer();
```
Comment (dylan-conway) [inline test/integration/sharp/sharp.test.ts]:
```suggestion
    const inputBuffer = await sharp(path.join(import.meta.dir, "bun.png")).resize(200, 200).toBuffer();
```
also for the other uses of `bun.png`

**Comment by dylan-conway** [comment]
Thanks for these tests. I tested locally with the lifecycle script branch and they are passing

File: test/integration/sharp/package.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    "sharp"
+  ],
+  "devDependencies": {
+    "bun-types": "latest"
+  },
+  "peerDependencies": {
+    "typescript": "^5.0.0"
+  },
+  "dependencies": {
+    "sharp": "^0.33.0"
```
Comment (dylan-conway) [inline test/integration/sharp/package.json]:
```suggestion
    "sharp": "0.33.0"
```

**Comment by ImBIOS** [comment]
@dylan-conway @paperdave updated

**Review by dylan-conway — APPROVED** [review]

**Comment by dylan-conway** [comment]
Thank you

**Pull request merged**